src/box/lua/config/utils/log.lua

```diff
@@ -0,0 +1,135 @@
+-- Logger wrapper with a few enhancements.
+--
+-- 1. Encode tables into JSON.
+-- 2. Enable all the messages on TT_CONFIG_DEBUG=1.
+--
+-- Hopefully, the wrapper can be dropped when the following logger
+-- problems will be solved.
+--
+-- * JSONify tables at logging (gh-8611)
+-- * Follow TT_LOG_LEVEL before log.cfg()/box.cfg() (gh-8092)
+--
+-- A solution of the following problems may simplify the code of
+-- this wrapper.
+--
+-- * mylog.cfg() configures the main logger (gh-8610)
+-- * No easy way to determine if a message should go to the log
+--   (gh-8730)
+
+local json_noexc = require('json').new()
+json_noexc.cfg({encode_use_tostring = true})
+
+local logger_name = 'tarantool.config'
+local log = require('log').new(logger_name)
+
+local func2level = {
+    [log.error] = 2,
+    [log.warn] = 3,
+    [log.info] = 5,
+    [log.verbose] = 6,
+    [log.debug] = 7,
+}
+
+local func2prefix = {
+    [log.error] = 'E> ',
+    [log.warn] = 'W> ',
+    [log.info] = 'I> ',
+    [log.verbose] = 'V> ',
+    [log.debug] = 'D> ',
+}
+
+local str2level = {
+    ['fatal'] = 0,
+    ['syserror'] = 1,
+    ['error'] = 2,
+    ['warn'] = 3,
+    ['crit'] = 4,
+    ['info'] = 5,
+    ['verbose'] = 6,
+    ['debug'] = 7,
+}
+
+local function say_closure(log_f)
+    local prefix = ''
+
+    -- Enable logging of everything if an environment variable is
+    -- set.
+    --
+    -- Useful for debugging.
+    --
+    -- Just setting of...
+    --
+    -- ```
+    -- TT_LOG_MODULES='{"tarantool.config": "debug"}'
+    -- ```
+    --
+    -- ...is not suitable due to gh-8092: messages before first
+    -- box.cfg() are not shown.
+    --
+    -- Explicit calling of...
+    --
+    -- ```
+    -- log.cfg({modules = {['tarantool.config'] = 'debug'}})
+    -- ```
+    --
+    -- is not suitable as well, because it makes the logger
+    -- already configured and non-dynamic options like
+    -- `box_cfg.log_nonblock` can't be applied at first box.cfg()
+    -- invocation.
+    --
+    -- So just prefix our log messages and use log.info().
+    local envvar = os.getenv('TT_CONFIG_DEBUG')
+    if envvar ~= nil and envvar ~= '0' and envvar ~= 'false' then
```

Comment:
I guess it is a thing of taste and there is no strict rule. As for me, it wouldn't improve readability.

Generally, if I need a multiline condition, I prefer to save it to a variable (since it is complex -- it should have a name) and make the if-condition one-liner. An indentation in the multiline conditions feels weird for me, at least when the language doesn't impose parentheses around. (It is a bit better, when the language does... but still looks ugly for me.)

Reply:
The same as in a previous comment (let's wait an opinion from a third reviewer, @igormunkin).

Reply:
I'd rather introduce a helper to handle environment variable values, add a table with the "falsy" values and iteratively checked against all the values in the table and yield the value of the desired type. This multiconditional if clause looks a bit unrelated here.

Furthermore, does 'FALSE' also work or not? And does numeric zero differ from the string one here? When this logic is enclosed in scope of a separate function, the code is definitely clearer.

Reply:
I didn't plan to polish it so much (this code is just a helper for development). But this is not a problem, done:

```diff
diff --git a/src/box/lua/config/utils/log.lua b/src/box/lua/config/utils/log.lua
index 3072c77ddc..ad5210d5e1 100644
--- a/src/box/lua/config/utils/log.lua
+++ b/src/box/lua/config/utils/log.lua
@@ -49,6 +49,24 @@ local str2level = {
     ['debug'] = 7,
 }
 
+-- Accept false/true case insensitively.
+--
+-- Accept 0/1 as boolean values.
+--
+-- If anything else is given, just return the default.
+local function boolean_fromenv(raw_value, default)
+    if raw_value == nil or raw_value == '' then
+        return default
+    end
+    if raw_value:lower() == 'false' or raw_value == '0' then
+        return false
+    end
+    if raw_value:lower() == 'true' or raw_value == '1' then
+        return true
+    end
+    return default
+end
+
 local function say_closure(log_f)
     local prefix = ''
 
@@ -81,8 +99,7 @@ local function say_closure(log_f)
     --
     -- The prefix represents the original log level of the
     -- message.
-    local envvar = os.getenv('TT_CONFIG_DEBUG')
-    if envvar ~= nil and envvar ~= '0' and envvar ~= 'false' then
+    if boolean_fromenv(os.getenv('TT_CONFIG_DEBUG'), false) then
         prefix = func2prefix[log_f]
         log_f = log.info
     end
```

Reply:
Thanks a lot! Look much better to be honest. Resolving.